packages/xo-server-sdn-controller/src/protocol/ovsdb-client.js

```diff
@@ -93,17 +101,49 @@ export class OvsdbClient {
       return
     }
 
-    const alreadyExist = await this._interfaceAndPortAlreadyExist(
+    const port = await this._interfaceAndPortAlreadyExist(
       bridge,
       remoteAddress,
       socket
     )
-    if (alreadyExist) {
-      socket.destroy()
-      this._adding = this._adding.filter(
-        elem => elem.id !== network.uuid || elem.addr !== remoteAddress
-      )
-      return bridge.name
+    if (port !== undefined) {
+      if (password === undefined) {
```

Comment:
Why not combine both these conditions?

Reply:
If `password === undefined` nothing to do (I return) if password is not undefined then I need to recreate the tunnel with the new password.

Reply:
IMHO, long functions with early `return`s (even more when they are nested) are a bad idea because it's difficult to understand the flow.